core/sail/api/src/main/java/org/eclipse/rdf4j/sail/SailConnection.java

```diff
@@ -187,6 +188,19 @@ default boolean hasStatement(Resource subj, IRI pred, Value obj, boolean include
 	 */
 	void begin(IsolationLevel level) throws UnknownSailTransactionStateException, SailException;
 
+	/**
+	 *
+	 * Override this method to receive the transaction settings. This method is always called (by the SailRepository)
+	 * before any .begin method is called, but may not be called if the sail is being used outside of the
+	 * SailRepository. Always remember to call super. receiveTransactionSettings(settings) if you override this method.
+	 *
+	 * @param settings the transaction settings on which this transaction operates. It may or may not contain the
+	 *                 isolation level.
+	 */
+	default void receiveTransactionSettings(TransactionSetting[] settings) {
```

Comment:
The other APIs use the `...` form. This call seems to be compatible with that so there doesn't seem to be a reason why it should use the `[]` form instead of `...`.

Reply:
Piggybacking off this comment: couldn't we just call this 'setTransactionSettings' rather than 'receive...'?

Reply:
I'm using array notation because this method isn't supposed to be called from user code. I prefer "receive" over "set" for methods that the developer is supposed to override in order to receive something.

Reply:
But not married to "receive", so can switch to "set" if you want.

Reply:
Let's go with 'set', and let's reword the javadoc slightly to go with that, something like this:

	/**
	 * Pass any transaction-specific settings to the Sail stack. This method needs to be called before the transaction is {@link #begin() started }.
	 * <p>
	 * Sail implementations can override this method to receive the transaction settings (to inspect and/or pass them 
	 * along to any wrapped sail objects). Remember to call <code>super.superTransactionSettings(settings)</code> 
	 * if you override this method.
	 *
	 * @param settings the transaction settings on which the next transaction operates. It may or may not contain the
	 *                 isolation level. 
	 * @since 3.3.0
	 */

I don't think we should say that only SailRepository is allowed to call it. That is no more or less true for this method than for anything else in the Sail API.

Reply:
As for the array vs vararg notation, I have a slight preference for a vararg for consistency, but can live with either. Your call.